src/forms/NumberInput.jsx

```diff
@@ -0,0 +1,184 @@
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value || '',
+		};
+
+		this.decrement = this.decrement.bind(this);
+		this.increment = this.increment.bind(this);
+		this.onBlur = this.onBlur.bind(this);
+		this.onChange = this.onChange.bind(this);
+		this.onFocus = this.onFocus.bind(this);
+	}
+
+	onBlur(e) {
+		if (document.activeElement !== this.fauxInputEl && document.activeElement !== this.decrementBtnEl && document.activeElement !== this.incrementBtnEl) {
+			this.fauxInputEl.classList.remove('focused');
+		}
+	}
+
+	onChange(e) {
+		this.setState({ value: e.target.value });
+		if (this.props.onChange) {
+			this.props.onChange(e);
+		}
+	}
+
+	onFocus(e) {
+		this.fauxInputEl.classList.add('focused');
+	}
+
+	increment() {
+		const currentVal = new Number(this.state.value);
+		const increment = new Number(this.props.step);
+		const newValue = currentVal + increment;
+
+		if (newValue > this.props.max) return;
+
+		this.setState({value: newValue});
+	}
+
+	decrement() {
+		const currentVal = new Number(this.state.value);
+		const decrement = new Number(this.props.step);
+		const newValue = currentVal - decrement;
+
+		if (newValue < this.props.min) return;
+
+		this.setState({value: newValue});
+	}
+
+	render() {
+		const {
+			children,
+			className,
+			error,
+			id,
+			label,
+			labelClassName,
+			max,
+			min,
+			name,
+			onBlur, // eslint-disable-line no-unused-vars
+			onFocus, // eslint-disable-line no-unused-vars
+			onChange, // eslint-disable-line no-unused-vars
+			required,
+			step,
+			value, // eslint-disable-line no-unused-vars
+			...other
+		} = this.props;
+
+		const classNames = {
+			field: cx(
+				'field--reset',
+				{ 'field--error': error },
+				className
+			),
+			label: cx(
+				{ required },
+				labelClassName
+			),
+			incrementBtn: cx(
+				'button--reset',
+				INCREMENT_BTN_CLASS
+			),
+			decrementBtn: cx(
+				'button--reset',
+				DECREMENT_BTN_CLASS
+			)
+		};
+
+		return (
+			<div>
+				<label className={classNames.label} htmlFor={id}>
+					{label}
+				</label>
+
+				<div
+					className='fauxInput'
+					ref={ el => this.fauxInputEl = el }>
+					<Flex align='center'>
+						<FlexItem>
+							<input type='number'
+								name={name}
+								max={max}
+								min={min}
+								step={step}
+								value={this.state.value}
+								required={required}
+								className={classNames.field}
+								onBlur={this.onBlur}
+								onFocus={this.onFocus}
+								onChange={this.onChange}
+								{...other} />
+						</FlexItem>
+
+						<FlexItem shrink>
+							<button
+								className={classNames.decrementBtn}
+								onBlur={this.onBlur}
+								onClick={this.decrement}
+								onFocus={this.onFocus}
+								ref={ el => this.decrementBtnEl = el }>
+								<Icon shape='minus' size='xs' />
+							</button>
+						</FlexItem>
+
+						<FlexItem shrink>
+							<button
+								className={classNames.incrementBtn}
+								onBlur={this.onBlur}
```

Comment:
@eilinora - The listeners aren't for styling the buttons, they're for setting a `focused` class on the fake input that wraps all of this stuff.

Let me know if I'm misunderstanding your feedback, or we can talk IRL :)